docker/images/mainnet/Dockerfile

```diff
@@ -1,50 +1,48 @@
 # skycoin build
 # reference https://github.com/skycoin/skycoin
-FROM golang:1.9 AS build
+FROM golang:1.10 AS build
 
-ENV SKYCOIN_VERSION=0.22.0
+ARG SKYCOIN_VERSION
 
-# dirs
-RUN mkdir -p $GOPATH/src/github.com/skycoin/skycoin
+COPY . $GOPATH/src/github.com/skycoin/skycoin
 
-# clone
-RUN wget -c https://github.com/skycoin/skycoin/archive/v${SKYCOIN_VERSION}.tar.gz
-
-# uncompress
-RUN tar zxf v${SKYCOIN_VERSION}.tar.gz -C $GOPATH/src/github.com/skycoin/skycoin/ --strip-components=1
-
-# install cli
-RUN cd $GOPATH/src/github.com/skycoin/skycoin/cmd/cli && \
-  CGO_ENABLED=0 GOOS=linux go build -a -installsuffix cgo -o /go/bin/skycoin-cli .
-
-# install skycoin
-RUN cd $GOPATH/src/github.com/skycoin/skycoin/cmd/skycoin && \
-  CGO_ENABLED=0 GOOS=linux go build -a -installsuffix cgo -o /go/bin/skycoin .
-
-# install address_gen
-RUN cd $GOPATH/src/github.com/skycoin/skycoin/cmd/address_gen && \
-  CGO_ENABLED=0 GOOS=linux go build -a -installsuffix cgo -o /go/bin/address_gen .
+# This code checks if SKYCOIN_VERSION is set and checkouts to that version if
+# so. The git stash line prevents the build to fail if there are any uncommited
+# changes in the working copy. It won't affect the host working copy.
+RUN sh -c \
+    'if test ${SKYCOIN_VERSION};then \
+        echo "Revision is set to: "${SKYCOIN_VERSION}; \
+        cd $GOPATH/src/github.com/skycoin/skycoin; \
+        git stash; \
+        git checkout ${SKYCOIN_VERSION}; \
+     fi'
 
+RUN cd $GOPATH/src/github.com/skycoin/skycoin && \
+  CGO_ENABLED=0 GOOS=linux go install -a -installsuffix cgo ./...
 
 # skycoin image
 FROM alpine:3.7
 
-ENV COIN="skycoin" \
-    RPC_ADDR="0.0.0.0:6430" \
-    DATA_DIR="/root/.$COIN" \
+ENV COIN="skycoin"
+ENV RPC_ADDR="0.0.0.0:6430" \
+    DATA_DIR="/data/.$COIN" \
     WALLET_DIR="/wallet" \
     WALLET_NAME="$COIN_cli.wlt"
 
 # copy all the binaries
 COPY --from=build /go/bin/* /usr/bin/
 
 # copy assets
-COPY --from=build /go/src/github.com/skycoin/skycoin/src/gui/static /usr/local/skycoin/static
+COPY --from=build /go/src/github.com/skycoin/skycoin/src/gui/static /usr/local/skycoin/src/gui/static
+
+# copy launcher
+COPY docker_launcher.sh /usr/local/bin
 
 # volumes
 VOLUME $WALLET_DIR
 VOLUME $DATA_DIR
 
 EXPOSE 6000 6420 6430
 
-CMD ["/usr/bin/skycoin", "--gui-dir=/usr/local/skycoin/static", "--web-interface-addr=0.0.0.0"]
+ENTRYPOINT ["docker_launcher.sh"]
+CMD ["--web-interface-addr=0.0.0.0", "--rpc-interface-addr=0.0.0.0", "--gui-dir=/usr/local/skycoin/src/gui/static"]
```

Comment:
Can't run with argument, node will crash when arguments are provided. 

Can put all in `ENTRYPOINT` like:
```dockerfile
ENTRYPOINT ["docker_launcher.sh", "--web-interface-addr=0.0.0.0", "--rpc-interface-addr=0.0.0.0", "--gui-dir=/usr/local/skycoin/src/gui/static"]
```

Reply:
Done